src/pages/workspaces/workspace/notebooks/NotebookLauncher.js

```diff
@@ -32,32 +74,250 @@ const NotebookLauncher = _.flow(
+      ['PREVIEW (READ-ONLY)']),
     Utils.cond(
-      [!canCompute, () => h(ButtonOutline, { onClick: () => setCopying(true) }, ['copy to another workspace to edit'])],
-      () => h(ButtonOutline, {
-        href: Nav.getLink('workspace-notebook-launch', { namespace, name, notebookName })
-      }, ['edit in Jupyter'])
+      [
+        !readOnlyAccess && clusterStatus === 'Creating', () => h(StatusMessage, { showSpinner: true }, [
+          'Creating notebook runtime environment, this will take 5-10 minutes. You can navigate away and return when it’s ready.'
+        ])
+      ],
+      [
+        !readOnlyAccess && clusterStatus === 'Starting', () => h(StatusMessage, { showSpinner: true }, [
+          'Starting notebook runtime environment, this may take up to 2 minutes.'
+        ])
+      ],
+      [
+        !readOnlyAccess && clusterStatus === 'Stopping', () => h(StatusMessage, { showSpinner: true }, [
+          'Notebook runtime environment is stopping. You can restart it after it finishes.'
+        ])
+      ],
+      [!readOnlyAccess && clusterStatus === 'Error', () => h(StatusMessage, ['Notebook runtime error.'])],
+      [
+        !readOnlyAccess && (clusterStatus === 'Running' || clusterStatus === 'Stopped' || clusterStatus === null),
+        () => h(Fragment, [
+          locked ? h(ButtonSecondary, {
+            style: { paddingRight: '1rem', paddingLeft: '1rem', paddingTop: '1rem', paddingBottom: '1rem', backgroundColor: colors.dark(0.1), height: '100%', marginRight: '2px' },
+            onClick: () => setFileInUseOpen(true)
+          }, [icon('lock', { style: { paddingRight: '3px' } }), 'EDIT (IN USE)']) : h(ButtonSecondary, {
+            style: { paddingRight: '1rem', paddingLeft: '1rem', paddingTop: '1rem', paddingBottom: '1rem', backgroundColor: colors.dark(0.1), height: '100%', marginRight: '2px' },
+            onClick: () => clusterActions('edit', clusterStatus)
+          }, [icon('edit', { style: { paddingRight: '3px' } }), 'EDIT']),
+          h(ButtonSecondary, {
+            style: { paddingRight: '1rem', paddingLeft: '1rem', backgroundColor: colors.dark(0.1), height: '100%', marginRight: '2px' },
+            onClick: () => {
+              hidePlaygroundModal === 'true' ? clusterActions('playground', clusterStatus) : setPlaygroundModalOpen(true)
+            }
+          }, [icon('chalkboard', { style: { paddingRight: '3px' } }), 'PLAYGROUND MODE']),
+          h(ButtonSecondary, {
+            style: { paddingRight: '1rem', paddingLeft: '1rem', backgroundColor: colors.dark(0.1), height: '100%' }
+          }, [
+            h(PopupTrigger, {
+              closeOnClick: true,
+              content: h(Fragment, [
+                h(MenuButton, { onClick: () => setCopyingNotebookName(notebookName) }, ['Make a Copy']),
+                h(MenuButton, { onClick: () => setExportingNotebookName(notebookName) }, ['Copy to another workspace']),
```

Comment:
nit: Since there's only one notebook on this page, these states could be booleans instead of the name.